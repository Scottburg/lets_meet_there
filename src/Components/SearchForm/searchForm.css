.form-container {
  display: flex;
  align-content: center;
  text-align: center;
  justify-content: center;
}
.form {
  display: flex;
  flex-flow: column nowrap;
  align-content: flex-end;
}
.formTitles {
  font-size: 0.8em;
  text-align: center;
}
.inputboxes {
  width: 100%;
}
.datePicker {
  margin: 15px 0px 15px 0px;
}
::placeholder {
  opacity: 0.4;
  font-size: 0.7em;
}
