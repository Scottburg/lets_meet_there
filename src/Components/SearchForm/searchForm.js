import React, { useState } from 'react';
import './searchForm.css';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'react-dates/lib/css/_datepicker.css';
import { useSelector } from 'react-redux';


const SearchForm = ({ searchFlights, getPlace }) => {
  const initialState = {
    from1: '',
    from2: '',
    currency: useSelector((state) => state.currency),
  };
  const [state, setState] = useState(initialState);
  const [focus, setFocus] = useState(null);

  const [dateRange, setdateRange] = useState({
    startDate: null,
    endDate: null,
  });
  const { startDate, endDate } = dateRange;
  const { from1, from2 } = state;

  const handleOnDateChange = (startDate, endDate) =>
    setdateRange(startDate, endDate);

  const handleChange = (event) => {
    const { name, value } = event.target;
    setState((prevState) => ({ ...prevState, [name]: value }));
  };

  const handleSubmit = async (event) => {
    event.preventDefault();
    const { from1, from2, currency } = state;
    if (from1 && from2 && dateRange) {
      try {
        const fmtFrom1 = await getPlace(from1);
        const fmtFrom2 = await getPlace(from2);
        const departDate = dateRange.startDate.format('YYYY-MM-DD');
        const returnDate = dateRange.endDate.format('YYYY-MM-DD');
        searchFlights(fmtFrom1, fmtFrom2, departDate, returnDate, currency);
        setState(initialState);
      } catch (e) {
        console.log(e);
      }
    } else {
      alert('Please fill in all fields');
    }
  };
  return (
    <div className="form-container">
      <div className="form">
        <form className="form" onSubmit={handleSubmit}>
          <div className="from1">
            <div className="formTitles">From</div>
            <label htmlFor="from1"></label>
            <input
              className="inputboxes"
              type="text"
              name="from1"
              onChange={handleChange}
              value={from1}
              placeholder="E.g. London"
            />
          </div>
          <div className="from2">
            <div className="formTitles">And</div>
            <label htmlFor="from2"></label>
            <input
              className="inputboxes"
              type="text"
              name="from2"
              onChange={handleChange}
              value={from2}
              placeholder="E.g. Barcelona"
            />
          </div>
          <div className="datePicker">
            <DateRangePicker
              startDatePlaceholderText="Depart"
              startDate={startDate}
              onDatesChange={handleOnDateChange}
              endDatePlaceholderText="Return"
              endDate={endDate}
              numberOfMonths={1}
              displayFormat="MMM D"
              showClearDates={true}
              focusedInput={focus}
              onFocusChange={(focus) => {
                setFocus(focus);
              }}
              startDateId="startDate"
              endDateId="endDate"
              minimumNights={0}
            />
          </div>
          <div className="buttonDiv">
            <button type="submit"> Lets Meet There!</button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default SearchForm;
