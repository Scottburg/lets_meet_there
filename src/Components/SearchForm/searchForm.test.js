import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import SearchForm from './searchForm';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

it('render correctly text component', () => {
  const searchForm = renderer.create(<SearchForm />).toJSON();
  expect(searchForm).toMatchSnapshot();
});

it('display correct text', () => {
  const searchForm = render(<SearchForm />);
  searchForm.findAllByText('Lets Meet There').then((data) => data);
});

it('should have an two text fields', () => {
  const searchForm = shallow(<SearchForm />);
  expect(searchForm.find('input[type="text"]').length).toEqual(2);
});

it('input fields should have correct props', () => {
  const searchForm = shallow(<SearchForm />);
  expect(searchForm.find('input[name="from1"]').props()).toEqual({
    onChange: expect.any(Function),
    type: 'text',
    className: 'inputboxes',
    name: 'from1',
    placeholder: 'E.g. London',
    value: '',
  });

  expect(searchForm.find('input[name="from2"]').props()).toEqual({
    onChange: expect.any(Function),
    type: 'text',
    className: 'inputboxes',
    name: 'from2',
    placeholder: 'E.g. Barcelona',
    value: '',
  });
});
