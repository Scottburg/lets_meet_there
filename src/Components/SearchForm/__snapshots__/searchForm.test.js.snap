// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly text component 1`] = `
<div
  className="form-container"
>
  <div
    className="form"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <div
        className="from1"
      >
        <div
          className="formTitles"
        >
          From
        </div>
        <label
          htmlFor="from1"
        />
        <input
          className="inputboxes"
          name="from1"
          onChange={[Function]}
          placeholder="E.g. London"
          type="text"
          value=""
        />
      </div>
      <div
        className="from2"
      >
        <div
          className="formTitles"
        >
          And
        </div>
        <label
          htmlFor="from2"
        />
        <input
          className="inputboxes"
          name="from2"
          onChange={[Function]}
          placeholder="E.g. Barcelona"
          type="text"
          value=""
        />
      </div>
      <div
        className="datePicker"
      >
        <div
          className="DateRangePicker DateRangePicker_1"
        >
          <div>
            <div
              className="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2 DateRangePickerInput__showClearDates DateRangePickerInput__showClearDates_3"
            >
              <div
                className="DateInput DateInput_1"
              >
                <input
                  aria-describedby="DateInput__screen-reader-message-startDate"
                  aria-label="Depart"
                  autoComplete="off"
                  className="DateInput_input DateInput_input_1"
                  disabled={false}
                  id="startDate"
                  name="startDate"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Depart"
                  readOnly={false}
                  required={false}
                  type="text"
                  value=""
                />
                <p
                  className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                  id="DateInput__screen-reader-message-startDate"
                >
                  Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                </p>
              </div>
              <div
                aria-hidden="true"
                className="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
                role="presentation"
              >
                <svg
                  className="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
                  focusable="false"
                  viewBox="0 0 1000 1000"
                >
                  <path
                    d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
                  />
                </svg>
              </div>
              <div
                className="DateInput DateInput_1"
              >
                <input
                  aria-describedby="DateInput__screen-reader-message-endDate"
                  aria-label="Return"
                  autoComplete="off"
                  className="DateInput_input DateInput_input_1"
                  disabled={false}
                  id="endDate"
                  name="endDate"
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Return"
                  readOnly={false}
                  required={false}
                  type="text"
                  value=""
                />
                <p
                  className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
                  id="DateInput__screen-reader-message-endDate"
                >
                  Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
                </p>
              </div>
              <button
                aria-label="Clear Dates"
                className="DateRangePickerInput_clearDates DateRangePickerInput_clearDates_1 DateRangePickerInput_clearDates_default DateRangePickerInput_clearDates_default_2 DateRangePickerInput_clearDates__hide DateRangePickerInput_clearDates__hide_3"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="DateRangePickerInput_clearDates_svg DateRangePickerInput_clearDates_svg_1"
                  focusable="false"
                  viewBox="0 0 12 12"
                >
                  <path
                    d="M11.53.47a.75.75 0 0 0-1.061 0l-4.47 4.47L1.529.47A.75.75 0 1 0 .468 1.531l4.47 4.47-4.47 4.47a.75.75 0 1 0 1.061 1.061l4.47-4.47 4.47 4.47a.75.75 0 1 0 1.061-1.061l-4.47-4.47 4.47-4.47a.75.75 0 0 0 0-1.061z"
                    fillRule="evenodd"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="buttonDiv"
      >
        <button
          type="submit"
        >
           Lets Meet There!
        </button>
      </div>
    </form>
  </div>
</div>
`;
